src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -455,28 +453,14 @@ ChipError::StorageType pychip_DeviceController_SetWiFiCredentials(const char * s
 
 ChipError::StorageType pychip_DeviceController_CloseSession(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeid)
 {
-#if 0
-    //
-    // Since we permit multiple controllers per fabric and each is associated with a unique fabric index, closing a session
-    // requires us to do so across all controllers on the same logical fabric.
-    //
-    // TODO: Enable this and remove the call below to DisconnectDevice once #19259 is completed. This is because
-    //       OperationalDeviceProxy instances that are currently active will remain un-affected by this call and still
-    //       provide a valid SessionHandle in the OnDeviceConnected call later when we call DeviceController::GetConnectedDevice.
-    //       However, it provides a SessionHandle that is incapable of actually vending exchanges since it is in a defunct state.
-    //
-    //       For now, calling DisconnectDevice will at least just correctly de-activate a currently active OperationalDeviceProxy
-    //       instance and ensure that subsequent attempts to acquire one will correctly re-establish CASE on the fabric associated
-    //       with the provided devCtrl.
-    //
-    auto err = devCtrl->SessionMgr()->ForEachCollidingSession(ScopedNodeId(nodeid, devCtrl->GetFabricIndex()), [](auto *session) {
-        session->MarkAsDefunct();
+    devCtrl->SessionMgr()->ForEachMatchingSession(ScopedNodeId(nodeid, devCtrl->GetFabricIndex()), [](auto * session) {
```

Comment:
This should be `ForEachCollidingSession`, which was what was there before.

Reply:
And the comments should be re-instated to explain why.

Reply:
`ForEachCollidingSession` doesn't exist. What was the difference between `ForEachCollidingSession` and `ForEachMatchingSession`?

I have a PR ready to upload, but waiting for a reply to this one question before uploading since I don't want to put load on CI that is going to be throw away work

Reply:
Need to use `ForEachMatchingSessionOnLogicalFabric`